modules/cudf/src/table/concatenate.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cudf/concatenate.hpp>
+#include <node_cudf/table.hpp>
+
+namespace nv {
+
+Napi::Value Table::concat(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+
+  std::vector<Table::wrapper_t> tables = args[0];
+  rmm::mr::device_memory_resource* mr  = args[1];
```

Comment:
Do we want to check the args using `NODE_CUDA_EXPECT`? I don't think it would be possible to call it with anything else.

Reply:
Maybe check that `tables` is at least length 1?

Reply:
Sorry, just wanted to clarify, throw a `NODE_CUDA_EXPECT` if `tables.size() == 0`?

Reply:
Yes that's what I meant to propose. It shouldn't ever happen given the construction of `DataFame.concat` supplying `this` but it seems like a good precaution anyway (cc @trxcllnt)

Reply:
Spoke with Bryan on Friday about the difference between...

```
NODE_CUDF_EXPECT(col->size() == num_rows_, "All Columns must be of same length");

vs
.
NAPI_THROW(Napi::Error::New(args.Env(),
    "Table constructor requires all columns to be of same length"));
```

Is there any reason why it seems they throw in two different ways? I can setup an example if needed, but these were the outputs...

```
NODE_CUDF_EXPECT(col->size() == num_rows_, "All Columns must be of same length");

terminate called after throwing an instance of 'cudf::logic_error'
  what():  cuDF failure:
All Columns must be of same length
    at ../../src/table.cpp:89
Aborted (core dumped)
```

```
NAPI_THROW(Napi::Error::New(args.Env(),
    "Table constructor requires all columns to be of same length"));

    Table constructor requires all columns to be of same length
      753 |     const b = Series.new([5, 6, 7, 8, 9]);
      754 |
    > 755 |     const result = new Table({columns: [a._col, b._col]});
          |                    ^
      756 |     console.log(result);
      757 |   });
      758 | });
      at Object.<anonymous> (test/cudf-data-frame-tests.ts:755:20)
```

Reply:
Solved, `args.Env()` needs to be passed in.

`NODE_CUDF_EXPECT(col->size() == num_rows_, "All Columns must be of same length", args.Env());`